### DISCUSSION TITLE
Building macos_x86_64 and macos_arm64 with gfortran support

### DISCUSSION THREAD

**Discussion by ragonneau** [pypa/cibuildwheel#1051] [Q&A] [answered]
I am building a [Python library](https://github.com/pdfo/pdfo) using `cibuildwheel` and `gfortran` for macOS on GitHub Actions. Since they do not provide ARM architectures of macOS, I have to build it using cross-compilation. To install gfortran, I use the [MacPython](https://github.com/MacPython/gfortran-install) installer, but the wheel repairing stage fails because the Fortran libraries are not found.

Did anyone face the same issue and solved it?
Thanks!

**Comment by MrBitBucket** [comment, marked as answer]
I am having the same issue with a python extension. I can build the required libs/includes offline and get the setup.py to download those as resources. Then I am trying to use https://cibuildwheel.readthedocs.io/en/stable/faq/#macos-passing-dyld_library_path-to-delocate
to make things work. I eventually put the lib/include folders underneath /tmp/mycache and that allowed the M1 wheel to include the arm64 dylib.

**Reply by ragonneau** [reply]
Brilliant, thank you very much! It was the final key to the answer. I installed `gfortran` using the [gfortran-install](https://github.com/MacPython/gfortran-install) repository. I then linked manually the DYLD library to `/usr/local/gfortran/lib` and I placed in my `pyproject.toml` the following code:
```
[tool.cibuildwheel.macos]
archs = "x86_64 arm64"
test-skip = "*_arm64 *_universal2:arm64"
environment = { CFLAGS="-std=c99 -fno-strict-aliasing", CC="clang", CXX = "clang++" }
repair-wheel-command = [
    "DYLD_LIBRARY_PATH=/usr/local/gfortran/lib delocate-listdeps {wheel}",
    "DYLD_LIBRARY_PATH=/usr/local/gfortran/lib delocate-wheel --require-archs {delocate_archs} -w {dest_dir} {wheel}"
]
```
